presto-main/src/main/java/com/facebook/presto/operator/scalar/MathFunctions.java

```diff
@@ -761,6 +762,32 @@ public static double chiSquaredCdf(
         return distribution.cumulativeProbability(value);
     }
 
+    @Description("Inverse of Poisson cdf given lambda (mean) parameter and probability")
+    @ScalarFunction
+    @SqlType(StandardTypes.INTEGER)
+    public static long inversePoissonCdf(
+            @SqlType(StandardTypes.DOUBLE) double lambda,
+            @SqlType(StandardTypes.DOUBLE) double p)
+    {
+        checkCondition(p >= 0 && p < 1, INVALID_FUNCTION_ARGUMENT, "p must be in the interval [0, 1)");
+        checkCondition(lambda > 0, INVALID_FUNCTION_ARGUMENT, "lambda must be greater than 0");
+        PoissonDistribution distribution = new PoissonDistribution(lambda);
+        return distribution.inverseCumulativeProbability(p);
+    }
+
+    @Description("Poisson cdf given the lambda (mean) parameter and value")
+    @ScalarFunction
+    @SqlType(StandardTypes.DOUBLE)
+    public static double poissonCdf(
+            @SqlType(StandardTypes.DOUBLE) double lambda,
+            @SqlType(StandardTypes.INTEGER) long value)
+    {
+        checkCondition(value >= 0, INVALID_FUNCTION_ARGUMENT, "value must be a non-negative integer");
+        checkCondition(lambda > 0, INVALID_FUNCTION_ARGUMENT, "lambda must be greater than 0");
```

Comment:
Since the library API is already doing this check, I say just do a try/catch and throw user_error

Reply:
Interesting. Since this is the method used by all the distribution functions (i.e.: normal, beta, chi-square, binomial), do you think it should be changed there as well?
If so - could you please help with this change? (I'm not experienced in Java, so want to make sure I understand what you're proposing and which error will be thrown)

Reply:
Just do something like: 
```
try { ... } catch(NotStrictlyPositiveException notStrictlyPositiveException) { throw new PrestoException(GENERIC_USER_ERROR, ...)
```

Look at StandardErrorCodes.java and other files to see the pattern what they do.

Reply:
And yeah it will be good to do for all of them.